v8go.cc

```diff
@@ -989,6 +996,20 @@ void ObjectSetIdx(ValuePtr ptr, uint32_t idx, ValuePtr val_ptr) {
   obj->Set(local_ctx, idx, prop_val->ptr.Get(iso)).Check();
 }
 
+int ObjectSetInternal(ValuePtr ptr, uint32_t idx, ValuePtr val_ptr) {
+  LOCAL_OBJECT(ptr);
+  m_value* prop_val = static_cast<m_value*>(val_ptr);
+
+  if (obj->InternalFieldCount() <= idx) {
+    return 0;
+  }
+
+  obj->SetAlignedPointerInInternalField(idx, nullptr);
```

Comment:
The documentation for SetAlignedPointerInInternalField says

> To retrieve such a field, GetAlignedPointerFromInternalField must be used, everything else leads to undefined behavior.

and there are no calls to GetAlignedPointerFromInternalField, so why is this needed?  If it is needed, then this line could use a comment clarifying its purpose and we could also use a test that would fail if someone tried to remove this line.

Reply:
I should definitely add a comment. When I worked on this, I couldn't set an internal field unless I called this. And what I mean is that if I attempted to call `SetInternalField` without calling this, it would crash. I did some searching online and saw these two being used together, so I gave it a try and it worked.

That said, I'm open to correcting it, if it's not correct.

Reply:
I'm not saying it isn't correct.  All I know is that removing this line doesn't cause a test to fail and that the purpose of the line wasn't clear when based on reading the v8 documentation for it.  If you could reproduce the crash in the test suite, it would be very helpful as a regression test, even if you don't understand why/how the line is needed to fix the crash.

Reply:
Absolutely! I'm very curious, though. If it doesn't crash, this can definitely be removed, but I can't quite remember what exactly the issue was. I may have some notes, so I'll take a look.

Thanks for pointing this out.

Reply:
I've removed this because I can't repro.